app/scripts/modules/amazon/src/function/configure/Network.tsx

```diff
@@ -0,0 +1,246 @@
+import * as React from 'react';
+import { Option } from 'react-select';
+import { IPromise } from 'angular';
+import { Observable, Subject } from 'rxjs';
+import { forOwn } from 'lodash';
+
+import {
+  Application,
+  FormikFormField,
+  IWizardPageComponent,
+  IVpc,
+  ISubnet,
+  HelpField,
+  IAccount,
+  IRegion,
+  ReactSelectInput,
+  TetheredSelect,
+  SubnetReader,
+  ReactInjector,
+  ISecurityGroupsByAccountSourceData,
+} from '@spinnaker/core';
+import { FormikErrors, FormikProps } from 'formik';
+import { IAmazonFunctionUpsertCommand } from 'amazon/index';
+import { VpcReader } from 'amazon/vpc';
+
+export interface ISubnetOption {
+  subnetId: string;
+  vpcId: string;
+}
+
+export interface INetworkProps {
+  formik: FormikProps<IAmazonFunctionUpsertCommand>;
+  isNew?: boolean;
+  app: Application;
+}
+
+export interface INetworkState {
+  vpcOptions: Array<{}>;
+  accounts: IAccount[];
+  regions: IRegion[];
+  subnets: ISubnetOption[];
+  securityGroups: ISecurityGroupsByAccountSourceData;
+}
+
+export class Network extends React.Component<INetworkProps, INetworkState>
+  implements IWizardPageComponent<IAmazonFunctionUpsertCommand> {
+  constructor(props: INetworkProps) {
+    super(props);
+    this.getAllVpcs();
+  }
+
+  public state: INetworkState = {
+    vpcOptions: [],
+    accounts: null,
+    regions: [],
+    subnets: [],
+    securityGroups: null,
+  };
+  private props$ = new Subject<INetworkProps>();
+  private destroy$ = new Subject<void>();
+
+  private getAllVpcs(): void {
+    VpcReader.listVpcs().then(Vpcs => {
+      this.state.vpcOptions = Vpcs;
+    });
+  }
+
+  public validate(): FormikErrors<IAmazonFunctionUpsertCommand> {
+    return {};
+  }
+
+  private getAvailableSubnets(): IPromise<ISubnet[]> {
+    return SubnetReader.listSubnetsByProvider('aws');
+  }
+
+  private getAvailableSecurityGroups(): IPromise<ISecurityGroupsByAccountSourceData> {
+    return ReactInjector.securityGroupReader.getAllSecurityGroups();
+  }
+  private makeSubnetOptions(availableSubnets: ISubnet[]): ISubnetOption[] {
+    const subOptions: ISubnetOption[] = [];
+    availableSubnets.forEach(s => {
+      const opt: ISubnetOption = {
+        subnetId: s.id,
+        vpcId: s.vpcId,
+      };
+      subOptions.push(opt);
+    });
+    // we have to filter out any duplicate options
+    const uniqueSubOptions = Array.from(new Set(subOptions.map(a => a.subnetId))).map(subnetId => {
+      return subOptions.find(a => a.subnetId === subnetId);
+    });
+    return uniqueSubOptions;
+  }
+
+  public componentDidUpdate() {
+    this.props$.next(this.props);
+  }
+
+  public componentWillUnmount(): void {
+    this.destroy$.next();
+  }
+
+  public componentDidMount(): void {
+    const subnets$ = Promise.resolve(this.getAvailableSubnets())
```

Comment:
Since `subnets$` is not an observable, it shouldn't get the `$` suffix.  Alternatively, you can use `Observable.fromPromise()`

Reply:
Renamed `subnets$` to `allSubnets`.
Change will appear in the next push.